toffy/qc_comp.py

```diff
@@ -246,209 +248,163 @@ def compute_99_9_intensity(image_data):
+            )
 
-            # STEP 2: extract non-zero mean intensity
-            nonzero_mean_intensity[i, j] = compute_nonzero_mean_intensity(image_data_np)
+        # STEP 2: extract non-zero mean intensity
+        nonzero_mean_intensity[i] = compute_nonzero_mean_intensity(image_data_np)
 
-            # STEP 3: extract total intensity
-            total_intensity[i, j] = compute_total_intensity(image_data_np)
+        # STEP 3: extract total intensity
+        total_intensity[i] = compute_total_intensity(image_data_np)
 
-            # STEP 4: take 99.9% value of the data and assign
-            intensity_99_9[i, j] = compute_99_9_intensity(image_data_np)
+        # STEP 4: take 99.9% value of the data and assign
+        intensity_99_9[i] = compute_99_9_intensity(image_data_np)
 
-    # convert the numpy arrays to pandas DataFrames
-    df_nonzero_mean_batch = pd.DataFrame(
-        nonzero_mean_intensity, columns=chans
+    # define a pandas DataFrame for each metric
+    df_nonzero_mean = pd.DataFrame(
+        columns=['fov', 'channel', 'Non-zero mean intensity'], dtype=object
     )
-
-    df_total_intensity_batch = pd.DataFrame(
-        total_intensity, columns=chans
+    df_total_intensity = pd.DataFrame(
+        columns=['fov', 'channel', 'Total intensity'], dtype=object
     )
-
-    df_99_9_intensity_batch = pd.DataFrame(
-        intensity_99_9, columns=chans
+    df_99_9_intensity = pd.DataFrame(
+        columns=['fov', 'channel', '99.9 intensity value'], dtype=object
     )
 
-    # append the batch_names as fovs to each DataFrame
-    df_nonzero_mean_batch['fov'] = fovs
-    df_total_intensity_batch['fov'] = fovs
-    df_99_9_intensity_batch['fov'] = fovs
+    # append the stats
+    df_nonzero_mean['Non-zero mean intensity'] = nonzero_mean_intensity
+    df_total_intensity['Total intensity'] = total_intensity
+    df_99_9_intensity['99.9 intensity value'] = intensity_99_9
 
-    # create a dictionary mapping the metric name to its respective DataFrame
-    qc_data_batch = {
-        'nonzero_mean_batch': df_nonzero_mean_batch,
-        'total_intensity_batch': df_total_intensity_batch,
-        '99_9_intensity_batch': df_99_9_intensity_batch
-    }
+    # append the FOV name and channel name
+    df_nonzero_mean['fov'] = fov_name
+    df_total_intensity['fov'] = fov_name
+    df_99_9_intensity['fov'] = fov_name
 
-    return qc_data_batch
+    df_nonzero_mean['channel'] = chans
+    df_total_intensity['channel'] = chans
+    df_99_9_intensity['channel'] = chans
 
+    # save the data to the qc_dir
+    df_nonzero_mean.to_csv(
+        os.path.join(bin_file_path, '%s_nonzero_mean_stats.csv') % fov_name, index=False
+    )
+    df_total_intensity.to_csv(
+        os.path.join(bin_file_path, '%s_total_intensity_stats.csv') % fov_name, index=False
+    )
+    df_99_9_intensity.to_csv(
+        os.path.join(bin_file_path, '%s_percentile_99_9_stats.csv') % fov_name, index=False
```

Comment:
Can the df manipulation be condensed into a for loop? We're doing the same operation on all three dfs